app/controllers/application_controller.rb

```diff
@@ -2,11 +2,32 @@
 class ApplicationController < ActionController::Base
   include Authentication
   include Pundit::Authorization
+  before_action :store_user_location!
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
+  def not_found
+    render file: Rails.public_path.join('404.html'), status: :internal_server_error, layout: false
```

Comment:
IMHO, 404 status code go with "Not Found", if you are throwing internal_server_error, you should use 500. What's your opinion on that?

Reply:
Yes bhaiya, You are right. I mistakenly added this status. I resolved it.
Thank you.